lib/textris/message.rb

```diff
@@ -81,15 +81,20 @@ def parse_from(from)
     end
 
     def parse_from_dual(from)
-      if (matches = from.to_s.match(/(.*)\<(.*)\>\s*$/).to_a).size == 3 &&
-          Phony.plausible?(matches[2])
+      matches = from.to_s.match(/(.*)\<(.*)\>\s*$/).to_a
+      return if matches.size != 3
+      if Phony.plausible?(matches[2]) || PhoneFormatter.is_a_short_code?(matches[2])
```

Comment:
maybe some simple helper method, that will be readable for getting name and number from the dual field